it/controllers/pensions/annualAllowance/PensionProviderPaidTaxControllerISpec.scala

```diff
@@ -0,0 +1,366 @@
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(pensionSummaryUrl(taxYearEOY))
+      }
+    }
+  }
+
+  ".submit" should {
+    userScenarios.foreach { user =>
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+
+        s"return $BAD_REQUEST error when no value is submitted" which {
+          lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> "")
+
+          lazy val result: WSResponse = {
+            dropPensionsDB()
+            authoriseAgentOrIndividual(user.isAgent)
+            insertCyaData(aPensionsUserData, aUserRequest)
+            urlPost(fullUrl(pensionProviderPaidTaxUrl(taxYearEOY)), body = form, follow = false, welsh = user.isWelsh,
+              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          "has the correct status" in {
+            result.status shouldBe BAD_REQUEST
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+          import Selectors._
+          import user.commonExpectedResults._
+          titleCheck(user.specificExpectedResults.get.expectedErrorTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY), captionSelector)
+          textOnPageCheck(expectedParagraph, paragraphSelector(2))
+          textOnPageCheck(user.specificExpectedResults.get.expectedYouCanFindThisOut, paragraphSelector(3))
+          radioButtonCheck(yesText, 1, checked = Some(false))
+          radioButtonCheck(expectedNoButAgreedToPayRadio, 2, checked = Some(false))
+          radioButtonCheck(noText, 3, checked = Some(false))
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(pensionProviderPaidTaxUrl(taxYearEOY), formSelector)
+          errorSummaryCheck(user.specificExpectedResults.get.expectedError, Selectors.yesSelector)
+          errorAboveElementCheck(user.specificExpectedResults.get.expectedError, Some("value"))
+        }
+      }
+    }
+
+    "redirect and update question to 'Yes' when user selects yes when there is no cya data" which {
+      lazy val form: Map[String, String] = Map(PensionProviderPaidTaxQuestionForm.yesNo -> PensionProviderPaidTaxQuestionForm.yes)
+
+      lazy val result: WSResponse = {
+        dropPensionsDB()
+        authoriseAgentOrIndividual(isAgent = false)
+        urlPost(fullUrl(pensionProviderPaidTaxUrl(taxYearEOY)), body = form, follow = false,
+          headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      "has a SEE_OTHER(303) status" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(pensionSummaryUrl(taxYearEOY))
+      }
+
+      "updates pensionProvidePaidAnnualAllowanceQuestion to Some(Yes)" in {
+        lazy val cyaModel = findCyaData(taxYearEOY, aUserRequest).get
```

Comment:
Can you add a todo to redirect to "how much tax did your pension provider pay" amount page